### DISCUSSION TITLE
Running R scripts on Mox

### DISCUSSION THREAD

**Discussion by yaaminiv** [RobertsLab/resources#1176] [Q&A] [answered]
Based on advice from [this discussion](https://github.com/RobertsLab/resources/discussions/1137), I'm going to run my `methylKit` R scripts on Mox. I have a few questions about running R scripts on Mox that I can't find the answer to [in our lab's Mox guide](https://robertslab.github.io/resources/mox_Running-a-Job/).

1. Is there a list of R packages that have been installed on Mox, or do I need to install every new R run?
2. If I run a command (ex. `head(myDataframe)`), does that output get written to the slurm-out file?
3. Looking the Hyak documentation, I can run [R](https://wiki.cac.washington.edu/display/hyakusers/Hyak+R+programming) or [R Studio](https://wiki.cac.washington.edu/display/hyakusers/Hyak+RStudio) interactively. Does anyone have experience with this/what are the pros and cons of running an interactive node vs. using a SBATCH script?

**Comment by yaaminiv** [comment, marked as answer]
> Is there a list of R packages that have been installed on Mox, or do I need to install every new R run?

Answered my own question based on [Hyak documentation](https://wiki.cac.washington.edu/display/hyakusers/Hyak+R+programming). I can install and save packages in my `gscratch` directory

**Comment by yaaminiv** [comment]
I created [this script](https://github.com/RobertsLab/project-gigas-oa-meth/blob/master/code/06-methylKit-mox.sh) to run R on mox based on a suggestion from @kubu4 in [this discussion](https://github.com/RobertsLab/resources/discussions/1137#discussioncomment-451811). When I ran the script on mox I got the following error:

<img width="589" alt="Screen Shot 2021-04-21 at 9 27 29 PM" src="https://user-images.githubusercontent.com/22335838/115655566-637a3d80-a2e8-11eb-8db1-35097e82e56d.png">

I'm not sure what it means, and I don't see this information in the [hyak documentation](https://wiki.cac.washington.edu/display/hyakusers/Hyak+R+programming). 

Where would I specify the above arguments, and what exactly are they? Alternatively, I could move my R code to a separate file, and call it within a SLURM script.

*edited because I was missing a sentence*

**Reply by kubu4** [reply]
Sorry! Change the shebang to:

`#!/gscratch/srlab/programs/R-3.6.2/bin/Rscript`

**Reply by kubu4** [reply]
And, alternatively, you could stick with the documentation provided. With that said, here's better documentation:

https://www.chpc.utah.edu/documentation/software/r-language.php

Specifically the section titled, "Runing a R batch script on the cluster (using SLURM)".

**Reply by yaaminiv** [reply]
Thanks! That worked and now I have new issues.

**Reply by kubu4** [reply]
Yay? 😆 

BTW, which worked: change shebang or calling `Rscript` in SLURM script?

**Reply by yaaminiv** [reply]
I changed the shebang and ran the script! But now I'm having trouble actually loading packages with `require()`. I think R doesn't know where to find the packages (I go into it in [this discussion](https://github.com/RobertsLab/resources/discussions/1187)).
